crates/formats/src/entryfilelist.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    char,
+    io::{self, Cursor, ErrorKind, Read, Seek},
+};
+
+use byteorder::{ReadBytesExt, LE};
+use flate2::read::ZlibDecoder;
+use thiserror::Error;
+use zerocopy::{FromBytes, FromZeroes, Ref, U32};
+
+use crate::io_ext::{ReadWidestringError, SeekExt};
+
+#[derive(Debug, Error)]
+pub enum EntryfilelistError {
+    #[error("Could not read string")]
+    String(#[from] ReadWidestringError),
+
+    #[error("Could not create reference to value")]
+    UnalignedValue,
+
+    #[error("Zlib error")]
+    Zlib,
+
+    #[error("Io error")]
+    Io(#[from] io::Error),
+}
+
+#[allow(unused)]
+pub struct EntryfilelistContainer<'a> {
+    bytes: &'a [u8],
+
+    header: &'a ContainerHeader,
+
+    compressed: &'a [u8],
+}
+
+impl<'a> EntryfilelistContainer<'a> {
+    pub fn parse(bytes: &'a [u8]) -> Result<Self, EntryfilelistError> {
+        let (header, compressed) = Ref::<_, ContainerHeader>::new_from_prefix(bytes)
+            .ok_or(EntryfilelistError::UnalignedValue)?;
+
+        Ok(Self {
+            bytes,
+            header: header.into_ref(),
+            compressed,
+        })
+    }
+
+    fn hint_size(&self) -> usize {
+        self.header.decompressed_size.get() as usize
+    }
+
+    pub fn decompress(&self) -> Result<Entryfilelist, EntryfilelistError> {
```

Comment:
> Grabbing all the strings alone is pushing the 500mb rn and its still not done

Is that in reference to this?

Reply:
Nope that's the size of the text file generated by entryfilelist-test when piped to a file